File: crates/nux-apple-product-extension/smoke/product_extension_consumer.c
#include "nux_product_extension.h"

#include <stddef.h>

int main(void) {
  NuxFile *file = NULL;
  NuxCapiResult *result = NULL;
  NuxStatus status =
      nux_product_file_import_configured(NULL, 0, NULL, &file, &result);
  if (status == NUX_STATUS_OK || file != NULL) {
    return 1;
  }
  if (result != NULL) {
    (void)nux_capi_result_free(result);
  }
  return 0;
}

File: crates/nux-apple-product-extension/smoke/product_extension_consumer.swift
import NuxieRuntimeC

@main
enum ProductExtensionConsumer {
    static func main() {
        var file: OpaquePointer?
        var result: OpaquePointer?
        let status = nux_product_file_import_configured(
            nil,
            0,
            nil,
            &file,
            &result
        )
        precondition(status != NUX_STATUS_OK.rawValue)
        precondition(file == nil)
        if let result {
            _ = nux_capi_result_free(result)
        }
    }
}
